src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -52,6 +52,8 @@ public Tags(KeyValuePair<string, object?>[] keyValuePairs)
 
     public readonly bool Equals(Tags other)
     {
+        // Equality check is done without sorting, so the order of tags matters.
```

Comment:
not sure if others appreciate such code comments! Happy to remove if not required.